opt_out_http_api/tests/test_api.py

```diff
@@ -39,13 +41,29 @@ def api_delete(self, path):
     def api_count(self, path):
         return treq.get("%s%s" % (self.url, path), persistent=False)
 
-    def add_opt_out(self, address_type, address):
-        self.app.save_opt_out(address_type, address)
 
 # Tests
 
     @inlineCallbacks
     def test_opt_out_found(self):
+        def fixed_uuid():
+            return '36'
+        self.patch(uuid, 'uuid4', fixed_uuid)
+        resp = yield self.api_put("/optouts/msisdn/+273121100")
+        self.assertEqual(resp.code, 200)
+        data = yield resp.json()
+        self.assertEqual(data, {
+            "status": {
+                "code": 200,
+                "reason": "OK",
+            },
+            "opt_out": {
+                "id": "36",
+                "address_type": "msisdn",
+                "address": "+273121100"
+            },
+        })
```

Comment:
I would simplify this to:
```
existing_opt_out = self.backend.put("msisdn", "+273121100")
resp = yield self.api_call("/optouts/msisdn/+273121100")
self.assertEqual(resp.code, 200)
...
```
and then replace `"id": "36"` with `"id": existing_opt_out["id"]`. I think that makes the whole test clearer and shorter and avoid unnecessarily relying on other API calls working.